sample_project/Source/sample_project/Measure/Measure.cpp

```diff
@@ -0,0 +1,261 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "Measure.h"
+
+ // Sets default values
+AMeasure::AMeasure()
+{
+	PrimaryActorTick.bCanEverTick = true;
+
+	// Load the necessary assets
+	static ConstructorHelpers::FObjectFinder<UClass> WidgetAsset(TEXT("/Game/SampleViewer/Samples/Measure/Blueprints/UI_Measure.UI_Measure_C"));
+	if (WidgetAsset.Succeeded()) {
+		UIWidgetClass = WidgetAsset.Object;
+	}
+
+	static ConstructorHelpers::FObjectFinder<UStaticMesh> MeshAsset(TEXT("/Game/SampleViewer/Samples/Routing/Geometries/Cube.Cube"));
+	if (MeshAsset.Succeeded()) {
+		RouteMesh = MeshAsset.Object;
+	}
+
+}
+
+// Called when the game starts or when spawned
+void AMeasure::BeginPlay()
+{
+	Super::BeginPlay();
+
+	SetupInput();
+
+	// Make sure mouse cursor remains visible
+	APlayerController* PC = UGameplayStatics::GetPlayerController(GetWorld(), 0);
+	if (PC)
+	{
+		PC->bShowMouseCursor = true;
+		PC->bEnableClickEvents = true;
+	}
+
+	// Create the UI and add it to the viewport
+	if (UIWidgetClass)
+	{
+		UIWidget = CreateWidget<UUserWidget>(GetWorld(), UIWidgetClass);
+		if (UIWidget)
+		{
+			UIWidget->AddToViewport();
+			WidgetFunction = UIWidget->FindFunction(FName("SetDistance"));
+			UnitDropdown = (UComboBoxString*)UIWidget->GetWidgetFromName(TEXT("UnitDropDown"));
+		}
+	}
+	unitTxt = " m";
+	unit = UArcGISLinearUnit::CreateArcGISLinearUnit(EArcGISLinearUnitId::Meters);
+	currentUnit = m;
+}
+
+// Called every frame
+void AMeasure::Tick(float DeltaTime)
+{
+	Super::Tick(DeltaTime);
+
+}
+
+void AMeasure::SetupInput()
+{
+	InputComponent = NewObject<UInputComponent>(this);
+	InputComponent->RegisterComponent();
+
+	if (InputComponent)
+	{
+		InputComponent->BindAction("PlaceRoutePoint", IE_Pressed, this, &AMeasure::AddStop);
+		EnableInput(GetWorld()->GetFirstPlayerController());
+	}
+}
+void AMeasure::AddStop()
+{
+	float traceLength = 10000000.f;
+	FVector position;
+	FVector direction;
+	FHitResult hit;
+	FVector tangent;
+
+	APlayerController* PlayerController = UGameplayStatics::GetPlayerController(GetWorld(), 0);
+	PlayerController->DeprojectMousePositionToWorld(position, direction);
+
+	bool bTraceSuccess = GetWorld()->LineTraceSingleByChannel(hit, position,
+		position + traceLength * direction, ECC_Visibility, FCollisionQueryParams());
+
+	if (bTraceSuccess && hit.GetActor()->GetClass() == AArcGISMapActor::StaticClass() && hit.bBlockingHit)
+	{
+		FActorSpawnParameters SpawnParam = FActorSpawnParameters();
+		SpawnParam.Owner = this;
+
+		auto lineMarker = GetWorld()->SpawnActor<ARouteMarker>(ARouteMarker::StaticClass(), hit.ImpactPoint, FRotator(0.f), SpawnParam);
```

Comment:
It could be `0.0f` or just `0`
```suggestion
		auto lineMarker = GetWorld()->SpawnActor<ARouteMarker>(ARouteMarker::StaticClass(), hit.ImpactPoint, FRotator(0.0f), SpawnParam);
```